koku/masu/external/downloader/report_downloader_base.py

```diff
@@ -118,4 +119,13 @@ def _process_manifest_db_record(self, assembly_id, billing_start, num_of_files,
         with ReportManifestDBAccessor() as manifest_accessor:
             if num_of_files != manifest_entry.num_total_files:
                 manifest_accessor.update_number_of_files_for_manifest(manifest_entry)
+
+        # Set whether a provider should be classed as large based on report file count
+        # TODO do something similar for OCP, maybe based on cluster count (not necessarily here)
+        if num_of_files > settings.XL_REPORT_COUNT:
```

Comment:
Could we add the `is_large_customer` check here too?
```suggestion
        if num_of_files > settings.XL_REPORT_COUNT or is_large_customer(schema):
```

If so....

Getting the schema here is quite a mess. I would update the `ReportDownloader` to require all args to be keywords:
```
class ReportDownloader:
    def __init__(
        self,
        *, <<<<< add this
        customer_name,
```
Doing this would give us `customer_name` in the `kwargs` of `ReportDownloaderBase`, so we could get the schema like:
```
schema = kwargs.get("customer_name")
```
(yes, the customer-name is the schema)

Reply:
Yeh I thought about this, the problem is this is really just handling extremely large AWS providers right now. Doing this would result in all providers that are part of the XL unleash flag getting marked as large but not all of them actually are. For example the PNT account has 2 large providers but because we do unleash at the schema level all its AWS providers are classed as large when really they are small.

Your right that getting schema here is a fun time though.